engine/alternative-runtimes.md

```diff
@@ -0,0 +1,285 @@
+  ```
+
+> **Note**
+>
+> You can specify additional arguments when you register a runtime using the
+> configuration file, but not when you use the CLI flag.
+
+### Use containerd shims
+
+Docker Engine supports the use of containerd runtime shims.
+Using shims, you can use alternative runtimes without having to configure or
+restart the Docker daemon.
+
+To use a containerd shim, you must install the shim binary on `PATH` on the
+system where the Docker daemon is running. With the shim installed,
+you can select the shim with `docker run` using the shim's fully qualified name.
+Fully qualified shim names take the form of `io.containerd.<shim>.<version>`.
+
+The following `docker run` command uses the Kata Containers runtime, provided
+that the shim binary is installed on `PATH`.
+
+```console
+$ docker run --runtime io.containerd.kata.v2 hello-world
+```
+
+#### Configure shims
+
+If you need to pass additional configuration for a containerd shim, you can
+use the `runtimes` option in the daemon configuration file.
+
+1. Edit the daemon configuration file by adding a `runtimes` entry for the
+   shim you want to configure.
+
+   - Specify the fully qualified name for the runtime in `runtimeType` key
+   - Add your runtime configuration under the `options` key
+
+   ```json
+   {
+     "runtimes": {
+       "runsc": {
+         "runtimeType": "io.containerd.runsc.v1",
+         "options": {
+           "TypeUrl": "io.containerd.runsc.v1.options",
+           "ConfigPath": "/etc/containerd/runsc.toml"
+         }
+       }
+     }
+   }
+   ```
+
+2. Restart the Docker daemon.
+
+   ```console
+   # systemctl restart docker
+   ```
+
+3. Use the customized runtime using the `--runtime` flag for `docker run`.
+
+   ```console
+   $ docker run --runtime runsc ...
+   ```
+
+## Tutorials
+
+The following sections show examples on how to use alternative container
+runtimes. The examples cover the following runtimes:
+
+- [youki](#youki)
+- [Wasmtime](#youki)
+
+The example commands in these tutorials assume an Ubuntu or Debian system.
+
+### youki
+
+youki is a container runtime written in Rust.
+It claims to be faster and use less memory than runc,
+making it a good choice for resource-constrained environments.
+
+The following procedure shows how to install youki, configure Docker daemon,
+and use youki as the container runtime with Docker.
+
+1. Install youki and its dependencies.
+
+   Run the following commands with `sudo`, or as root.
+   This installs youki to `/usr/local/bin/youki`.
+
+   ```bash
+   #!/bin/bash
+
+   # Install the Rust toolchain
+   curl -sSf https://sh.rustup.rs | sh -s -- -y
+   source "$HOME/.cargo/env"
+
+   # Install build dependencies
+   apt-get install -y --no-install-recommends \
+       pkg-config \
+       libsystemd-dev \
+       libdbus-glib-1-dev \
+       build-essential \
+       libelf-dev \
+       libseccomp-dev \
+       libclang-dev \
+       libssl-dev
+
+   # Install youki
+   cargo install \
```

Comment:
It may be easier to get it from GitHub releases rather than using cargo. I hope this helps.

Reply:
The problem with the GitHub release is that it wouldn't work with Docker Desktop. It needs to include [this commit](https://github.com/containers/youki/commit/2ff8b97e676ebf848842df8fbc01f1b5c5c7b08c).
@utam0k would it be possible to cut another release?

Reply:
We intend to release it soon. I'll look for some time and release it this week.
https://github.com/containers/youki/issues/2039

Reply:
Thanks!